avm/res/app/managed-environment/main.bicep

```diff
@@ -0,0 +1,235 @@
+      dnsSuffix: dnsSuffix
+    }
+    vnetConfiguration: {
+      internal: internal
+      infrastructureSubnetId: !empty(infrastructureSubnetId) ? infrastructureSubnetId : null
+      dockerBridgeCidr: !empty(infrastructureSubnetId) ? dockerBridgeCidr : null
+      platformReservedCidr: empty(workloadProfiles) && !empty(infrastructureSubnetId) ? platformReservedCidr : null
+      platformReservedDnsIP: empty(workloadProfiles) && !empty(infrastructureSubnetId) ? platformReservedDnsIP : null
+    }
+    workloadProfiles: !empty(workloadProfiles) ? workloadProfiles : null
+    zoneRedundant: zoneRedundant
+    infrastructureResourceGroup: infrastructureResourceGroupName
+  }
+}
+
+resource managedEnvironment_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [for (roleAssignment, index) in (roleAssignments ?? []): {
+  name: guid(managedEnvironment.id, roleAssignment.principalId, roleAssignment.roleDefinitionIdOrName)
+  properties: {
+    roleDefinitionId: contains(builtInRoleNames, roleAssignment.roleDefinitionIdOrName) ? builtInRoleNames[roleAssignment.roleDefinitionIdOrName] : contains(roleAssignment.roleDefinitionIdOrName, '/providers/Microsoft.Authorization/roleDefinitions/') ? roleAssignment.roleDefinitionIdOrName : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName)
+    principalId: roleAssignment.principalId
+    description: roleAssignment.?description
+    principalType: roleAssignment.?principalType
+    condition: roleAssignment.?condition
+    conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+    delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+  }
+  scope: managedEnvironment
+}]
+
+resource managedEnvironment_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: managedEnvironment
+}
+
+@description('The name of the resource group the Managed Environment was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The location the resource was deployed into.')
+output location string = managedEnvironment.location
+
+@description('The name of the Managed Environment.')
+output name string = managedEnvironment.name
+
+@description('The resource ID of the Managed Environment.')
+output resourceId string = managedEnvironment.id
+
+@description('The Default domain of the Managed Environment.')
+output defaultDomain string = managedEnvironment.properties.defaultDomain
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type diagnosticSettingType = {
```

Comment:
This does not seem to be used

Reply:
This should be removed as `managedEnvironment` has its own properties to support console & system logs